internal/charmstore/migrations.go

```diff
@@ -222,3 +226,33 @@ func populateWriteACL(db StoreDatabase) error {
 	logger.Infof("%d base entities updated", counter)
 	return nil
 }
+
+// denormalizeEntityACLs updates all entities with an ACL copied from the
+// base entity. This function is not supposed to be called directly.
+func denormalizeEntityACLs(db StoreDatabase) error {
+	copyACLs := bson.JavaScript{
+		Code: jsACLCopyFunc,
+		Scope: bson.D{{
```

Comment:
Is it actually worth passing the scope here? Couldn't we just instantiate some JS that holds the collection names inline, given that the collection names never change?